action/protocol/staking/contractstake_indexer.go

```diff
@@ -51,3 +62,83 @@ func init() {
 		panic(err)
 	}
 }
+
+// NewDelayTolerantIndexer creates a delay tolerant indexer
+func NewDelayTolerantIndexer(indexer ContractStakingIndexer, duration time.Duration) ContractStakingIndexer {
+	return &delayTolerantIndexer{ContractStakingIndexer: indexer, duration: duration}
+}
+
+// NewDelayTolerantIndexerWithBucketType creates a delay tolerant indexer with bucket type
+func NewDelayTolerantIndexerWithBucketType(indexer ContractStakingIndexerWithBucketType, duration time.Duration) ContractStakingIndexerWithBucketType {
+	return &delayTolerantIndexerWithBucketType{&delayTolerantIndexer{ContractStakingIndexer: indexer, duration: duration}}
+}
+
+func (c *delayTolerantIndexer) wait(height uint64) (bool, error) {
+	// first check if the height is already reached
+	h, err := c.Height()
+	if err != nil {
+		return false, err
+	}
+	if h >= height {
+		return true, nil
+	}
+	// wait for the height to be reached
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+	timer := time.NewTimer(c.duration)
+	defer timer.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			h, err := c.Height()
+			if err != nil {
+				return false, err
+			}
+			if h >= height {
+				return true, nil
+			}
+		case <-timer.C:
+			return false, nil
+		}
+	}
+}
+
+func (c *delayTolerantIndexer) Buckets(height uint64) ([]*VoteBucket, error) {
+	_, err := c.wait(height)
```

Comment:
looks like the first return value `bool` is not used at all, then just don't return it?

Reply:
disscussed offline